src/main/java/tr/com/srdc/cda2fhir/transform/ResourceTransformerImpl.java

```diff
@@ -1948,6 +1948,15 @@ public EntryResult tMedicationActivity2MedicationStatement(MedicationActivity cd
 			fhirDosage.setMaxDosePerPeriod(dtt.tRTO2Ratio((RTO) cdaMedicationActivity.getMaxDoseQuantity()));
 		}
 
+		if (cdaMedicationActivity.getText() != null && !cdaMedicationActivity.getText().isSetNullFlavor()) {
+			String freeTextInstruction = dtt.tED2Annotation(cdaMedicationActivity.getText(),
```

Comment:
so I think this is a wrong, even though it probably works. If we are setting the sig to dose.text (which is fine), we should get the sig from where it's guaranteed to be a sig, it just happens to be in the text reference in our samples here.  That would on page 609 of the spec, that free text sig can come in. Page 598 says we can get a sig object in, so let's pull from that (it would work looking at the samples as well)

Reply:
Done